solidusio/solidus
# Fix Request Bodies in API Documentation

**Description**

As raised in #4058 and #3983, we are documenting all our `POST`/`PUT`/`PATCH` API endpoints wrongly.

We need to wrap the JSON of the request body with the object name that we are trying to create/edit. This is a example of the correct format, that we will have after this PR is merged:

```json
 {
  "product": {
    "name": "Product Name",
    "other": "fields"
   }
}

```

 But we are currently documenting all the endpoints to be used as:

```json
{
  "name": "Product Name",
  "other": "fields"
}
```

**Enpoints Fixed:**

- [x] variant
- [x] payment
- [x] address
- [x] line-item
- [x] order
- [x] coupon-code
- [x] shipment
- [x] zone
- [x] product-property
- [x] stock-location
- [x] taxon
- [x] taxonomy
- [x] property
- [x] image
- [x] user
- [x] return-authorization
- [x] customer-return
- [x] option-value
- [x] stock-item


**How to test this PR's result:**

Create a new project in Stoplight Studio, copy/paste the resulting oas.yml file, which should be [available here](https://github.com/nebulab/solidus/blob/kennyadsl/fix-api-documentation-inputs/api/openapi/solidus-api.oas.yml) and watch the preview for the result.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)
